### PR TITLE
testing/libneo4j-client: update to version 2.2.0

### PR DESCRIPTION
**Pull request by cleishm**

### CODE REVIEW

File: testing/libneo4j-client/APKBUILD [hunk 1]
```diff
@@ -21,17 +21,42 @@ build() {
...
     mkdir -p "$pkgdir"/usr/share/doc/libneo4j-client
     mv doc/html/* "$pkgdir"/usr/share/doc/libneo4j-client/
 }
 
+doc() {
+    depends="$depends_doc"
+    pkgdesc="$pkgdesc (documentation)"
+    install_if="docs $pkgname=$pkgver-r$pkgrel"
+
+    if [ "X$subpkgname" = "Xlibneo4j-client-doc" ]; then
```
Comment (jirutka) [inline testing/libneo4j-client/APKBUILD]: Oh my… please, this `x` non-sense in `[` is a workaround for a bug in one shell implementation that has been fixed several **decades** ago! It’s not needed anymore for very long time.

File: testing/libneo4j-client/APKBUILD [hunk 2]
```diff
@@ -21,17 +21,42 @@ build() {
...
+        mv "$pkgdir"/usr/share/man "$subpkgdir"/usr/share/
+        find "$subpkgdir"/usr/share/man/ -type f | xargs gzip -9
+    else
+        die "Unknown doc package $pkgname"
+    fi
+
+    # remove if empty, ignore error (not empty)
+    rmdir "$pkgdir"/usr/share "$pkgdir"/usr || true 2>/dev/null
+    return 0
+}
```
Comment (jirutka) [inline testing/libneo4j-client/APKBUILD]: This is all unnecessarily complicated. We usually don’t make multiple `-doc` subpackages from a single abuild. Why do you need it?

File: testing/libneo4j-client/APKBUILD (hunk 1 above)
Comment (cleishm) [reply testing/libneo4j-client/APKBUILD]: Old habits, happy to remove.

File: testing/libneo4j-client/APKBUILD (hunk 2 above)
Comment (cleishm) [reply testing/libneo4j-client/APKBUILD]: It's building two different binary packages - a library and a shell client. The library has documentation file (HTML) and the shell has a man page. So I put them in separate doc packages.

Happy to do so differently if you can advise your preference.

**Pull request closed without merging**